rust/kcl-lib/src/parsing/parser.rs

```diff
@@ -2058,6 +2059,52 @@ fn declaration(i: &mut TokenSlice) -> PResult<BoxNode<VariableDeclaration>> {
     }))
 }
 
+fn ty_decl(i: &mut TokenSlice) -> PResult<BoxNode<TypeDeclaration>> {
+    let (visibility, visibility_token) = opt(terminated(item_visibility, whitespace))
+        .parse_next(i)?
+        .map_or((ItemVisibility::Default, None), |pair| (pair.0, Some(pair.1)));
+
+    let decl_token = ty(i)?;
+    let start = visibility_token.map(|t| t.start).unwrap_or_else(|| decl_token.start);
+    whitespace(i)?;
+
+    let name = identifier(i)?;
+    let mut end = name.end;
+
+    let args = if peek(open_paren).parse_next(i).is_ok() {
```

Comment:
`kcl.grammar` needs to get updated with this.

Reply:
I only added the non-paranthesised form, since I couldn't wrestle the grammar into whatever kind of LALR whatevers is required (I hate declarative parsers so much).